### ISSUE TITLE
Reformatting a file with rustfmt doesn't work while reformatting a project does

### ISSUE THREAD

**Issue by mvlabat** [intellij-rust/intellij-rust#6980]
## Environment

* **IntelliJ Rust plugin version:** 0.3.143.3737-203
* **Rust toolchain version:** 1.52.0-nightly (107896c32 2021-03-15) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2020.3.2 (CL-203.7148.70)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

Basically, the title. Running `Reformat File with Rustfmt` action doesn't work, but `Reformat Cargo Project with Rustfmt` does.

## Steps to reproduce

I don't really know. It suddenly stopped working after I switched to the `nightly-2021-02-10` toolchain and then back to the latest one. I believe there was no `rustfmt` for the Windows version of `nightly-2021-02-10`.

**Comment by mchernyavsky** [comment]
I think the problem is this:

```bash
$ rustup default nightly-2021-02-10
info: syncing channel updates for 'nightly-2021-02-10-x86_64-apple-darwin'
info: latest update on 2021-02-10, rust version 1.52.0-nightly (097bc6a84 2021-02-09)
error: component 'rustfmt' for target 'x86_64-apple-darwin' is unavailable for download for channel nightly-2021-02-10
Sometimes not all components are available in any given nightly. If you don't need the component, you can remove it with:

    rustup component remove --toolchain nightly-2021-02-10 --target x86_64-apple-darwin rustfmt
```

`Reformat Cargo Project with Rustfmt` uses  [`cargo fmt`](https://github.com/rust-lang/rustfmt#running-cargo-fmt) .

**Comment by mvlabat** [comment]
@mchernyavsky yup, to me it looks suspicious and possibly related as well. And, as I said, I switched back to the newer one, where `rustfmt` exists, but it didn't start working in CLion after that anyway (and, again, it works for the whole project, but not for a file). Meanwhile, `cargo.exe fmt` works for me in `cmd`.

If there's any other debug info I can provide (logs, etc), I'll be glad to help.

**Comment by mvlabat** [comment]
Oh, I've tried to run `rustfmt` via `Ctrl+Alt+L` hotkey with enabled "Use rustfmt instead of built-in formatter", and I've got this error:

> Cannot run program "C:\Users\mvlabat\.cargo\bin\rustfmt.exe" (in directory "D:\Programming\muddle-run"): CreateProcess error=2, The system cannot find the file specified

I guess it shouldn't look for `rustfmt.exe` only in `C:\Users\mvlabat\.cargo\bin`.

**Comment by mvlabat** [comment]
Alright, changing "Toolchain location" in the settings to `C:\Users\mvlabat\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin` has fixed the problem for me. Not sure whether `C:\Users\mvlabat\.cargo\bin` is sensible enough and I'm also not sure how exactly it worked before downgrading with `rustup` and eventually breaking everything. Probably, my `rustfmt` was installed manually via `cargo` and I removed it in the process, which caused the confusion.

Though I guess the question about where to look for `rustfmt` is still open. Imo, for reformatting a specific file, the plugin should still rely on `cargo` to discover `rustfmt` location. It looks that it works like this only for reformatting the whole project, but for reformatting a specific file the plugin just tries to find `rustfmt` binary inside the toolchain dir, which doesn't work in all cases.
